### PR TITLE
ENG-8098: make authenticated conn cache aware of hashing.

### PR DESCRIPTION
**Pull request by akhanzode**

### CODE REVIEW

File: src/frontend/org/voltdb/HTTPClientInterface.java [hunk 1]
```diff
@@ -346,13 +347,13 @@ private AuthenticationResult getAuthenticationResult(Request request) {
 
         try {
             // get a connection to localhost from the pool
-            Client client = m_connections.get().getClient(username, password, hashedPasswordBytes, adminMode);
-            if (client != null) {
-                return new AuthenticationResult(client, adminMode, username, "");
+            Pair<Client, ClientAuthHashScheme> clientp = m_connections.get().getClient(username, password, hashedPasswordBytes, adminMode);
```
Comment (vtkstef) [inline src/frontend/org/voltdb/HTTPClientInterface.java]: Can we please please please please not make it a pair? A simple two member class is so much easier to read and understand. please please

File: src/frontend/org/voltdb/client/AuthenticatedConnectionCache.java [hunk 2]
```diff
@@ -211,10 +215,12 @@ public synchronized Client getClient(String userName, String password, byte[] ha
                 conn = null;
                 throw ioe;
             }
-            m_connections.put(admin ? userNameWithAdminSuffix : userName, conn);
+            synchronized (m_connections) {
```
Comment (vtkstef) [inline src/frontend/org/voltdb/client/AuthenticatedConnectionCache.java]: Why are we synchronizing here too when the method is already synchronized?

File: tests/frontend/org/voltdb/TestJSONInterface.java [hunk 3]
```diff
@@ -266,11 +271,16 @@ private static String httpUrlOverJSON(String method, String url, String user, St
...
         StringBuilder decodedString = new StringBuilder();
         String line;
-        while ((line = in.readLine()) != null) {
-            decodedString.append(line);
+        try {
+            while ((line = in.readLine()) != null) {
+                decodedString.append(line);
+            }
+        } catch (Exception ex) {
+            ex.printStackTrace();
```
Comment (vtkstef) [inline tests/frontend/org/voltdb/TestJSONInterface.java]: should we not fail here?

File: src/frontend/org/voltdb/client/AuthenticatedConnectionCache.java (hunk 2 above)
Comment (akhanzode) [inline src/frontend/org/voltdb/client/AuthenticatedConnectionCache.java]: my bad will remove

File: src/frontend/org/voltdb/HTTPClientInterface.java (hunk 1 above)
Comment (akhanzode) [inline src/frontend/org/voltdb/HTTPClientInterface.java]: sure with so many please I can change to a class or reuse Connection

File: tests/frontend/org/voltdb/TestJSONInterface.java (hunk 3 above)
Comment (akhanzode) [inline tests/frontend/org/voltdb/TestJSONInterface.java]: Sure. it does fail in test as return data is bad.

**Comment by akhanzode** [comment]
Can you look at it once again? Dont merge it I will do squash merge at the end so it can be cpicked.

File: src/frontend/org/voltdb/client/AuthenticatedConnectionCache.java [hunk 4]
```diff
@@ -211,10 +225,12 @@ public synchronized Client getClient(String userName, String password, byte[] ha
...
+                m_connections.put(ckey, conn);
+            }
             attemptToShrinkPoolIfNeeded();
```
Comment (vtkstef) [inline src/frontend/org/voltdb/client/AuthenticatedConnectionCache.java]: you may want to add the attempToShrinkPoolIfNeeded within the synchronized block

**Comment by vtkstef** [comment]
Ok I am all set. You may proceed with your squash

**Comment by akhanzode** [comment]
squash merged.

**Pull request closed without merging**